src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -267,11 +267,8 @@ protected void AssertSelf ()
 					// HttpClientHandler throws an exception in this instance, but I think it's not a good
 					// idea. We'll return the response message with all the information required by the
 					// application to fill in the blanks and provide the requested credentials instead.
-					//
-					// We should return the body of the response too but, alas, the Java client will throw
-					// a, wait for it, FileNotFound exception if we attempt to access the input stream. So
-					// no body, just a dummy. Java FTW!
-					ret.Content = new StringContent ("Unauthorized", Encoding.ASCII);
+
+					ret.Content = GetErrorContent(httpConnection,  new StringContent ("Unauthorized", Encoding.ASCII));
```

Comment:
Formatting - please follow the [Mono Coding Guidelines](http://www.mono-project.com/community/contributing/coding-guidelines/#style-guidelines), in this case there should be a space following the `GetErrorContent` call.